File: 240417/오목/O-mok.py
board = [list(map(int, input().split())) for _ in range(19)]

winner = 0
mid_coord = None

# 상, 우상, 우, 우하
dx = [-1, -1, 0, 1]
dy = [0, 1, 1, 1]

def DFS(x, y, cnt, color, f_dir):
    global winner, mid_coord

    if cnt == 5:
        winner = color
        mid_coord = (x - dx[f_dir] * 2, y - dy[f_dir] * 2)
        return

    mx, my = x + dx[f_dir], y + dy[f_dir]
    if 0 <= mx < 19 and 0 <= my < 19 and board[mx][my] == color:
        DFS(mx, my, cnt + 1, color, f_dir)
    else:
        return

for i in range(len(board)):
    for j in range(len(board[0])):
        if board[i][j] == 1 or board[i][j] == 2:
            for k in range(4):
                DFS(i, j, 1, board[i][j], k)

if winner != 0:
    print(winner)
    print(mid_coord[0] + 1, mid_coord[1] + 1)
else:
    print(0)

File: 260216/오목/O-mok.cpp
#include <bits/stdc++.h>
using namespace std;

int board[19][19];

int winner = 0;
pair<int,int> mid_coord = {-1, -1};

int dx[4] = {-1, -1, 0, 1};
int dy[4] = { 0,  1, 1, 1};

void DFS(int x, int y, int cnt, int color, int f_dir) 
{
    if (winner != 0) return; 

    if (cnt == 5) 
    {
        winner = color;
        mid_coord = {x - dx[f_dir] * 2, y - dy[f_dir] * 2};
        return;
    }

    int mx = x + dx[f_dir];
    int my = y + dy[f_dir];

    if (0 <= mx && mx < 19 && 0 <= my && my < 19 && board[mx][my] == color) 
    {
        DFS(mx, my, cnt + 1, color, f_dir);
    } 
    else 
    {
        return;
    }
}

int main()
{
    for (int i = 0; i < 19; i++) 
    {
        for (int j = 0; j < 19; j++) 
        {
            cin >> board[i][j];
        }
    }

    for (int i = 0; i < 19; i++) 
    {
        for (int j = 0; j < 19; j++) 
        {
            if (board[i][j] == 1 || board[i][j] == 2) 
            {
                for (int k = 0; k < 4; k++) 
                {
                    DFS(i, j, 1, board[i][j], k);
                }
            }
        }
    }

    if (winner != 0) 
    {
        cout << winner << "\n";
        cout << mid_coord.first + 1 << " " << mid_coord.second + 1 << "\n";
    } 
    else
    {
        cout << 0 << "\n";
    }

    return 0;
}
